core/src/main/java/at/roteskreuz/stopcorona/skeleton/core/utils/SharedPreferencesExtensions.kt

```diff
@@ -352,8 +360,15 @@ fun <T> SharedPreferences.observe(
                 emitter.onNext(Optional.ofNullable(sharedPreferences.retrieve(key, defaultValue)))
             }
         }
+
+        // Save from garbage collection
+        sharedPreferenceListeners.add(prefsListener)
```

Comment:
Wouldn't be enough to just have a runnable reference in a local variable inside the method? I guess closure principle should prevent to garbage collect it.
Meaning no static list is needed.

Reply:
I do not fully understand the the suggestion. If your only strong reference comes from a local variable then you can be garbage collected as soon as the variable goes out of scope. This is on return from the method.
If you have a cycle between closure and variable, that does not help either, as the whole cycle is not referenced from outside and can be garbage collected.